skbio/diversity/tests/test_driver.py

```diff
@@ -298,20 +298,20 @@ def test_invalid_input(self):
 
         # additional kwargs
         self.assertRaises(TypeError, beta_diversity, 'euclidean',
-                          [[0, 1, 3, 4], [0, 3, 12, 42]],
+                          [[0, 1, 3], [0, 3, 12]],
```

Comment:
Can you update these tests to use the following form:

```python
with six.assertRaisesRegex(self, TypeError, "regex that must match error message"):
    beta_diversity('euclidean', ...)
```

Using `six.assertRaisesRegex` ensures the correct exception type is raised and that the error message is the one you're expecting. Using the context manager is the preferred way of testing error cases because it is easier to read and to run the error-raising code (e.g., to view the actual error that's raised) b/c you can just comment out the `with` line and dedent.

Another note (not necessary to change in this PR unless you want to; we should have caught this during review): try to avoid test cases that test many different things. Most of the tests in this file are long and test several cases in a single test method. We've started using Arrange, Act, Assert (#1149) and it has really improved the quality of the unit tests.